RubeesEat/Pages/Index.cshtml

```diff
@@ -3,10 +3,10 @@
 @inject IBillRepository billRepository
 @inject IPersonRepository personRepository
 
-<h1>Hello, @personRepository.GetCurrentUser().FirstName!</h1>
+<h1>Hello, @personRepository.GetOrCreateUser(HttpContext.User).FirstName!</h1>
```

Comment:
Funnily enough, I actually already wrote a ticket for this yesterday. I originally mentioned it in a ticket for editing bills because there we use CurrentUser even more, and I discussed with Irem that we should make it into a seperate ticket. https://re-motion.atlassian.net/jira/software/projects/DSEAT/boards/17?selectedIssue=DSEAT-64

My original thought was to add it to the personRepo, so that when you use GetOrCreateUser, it first checks if it already queried it before. Should we do it in the controller instead?

Reply:
Good catch on the ticket! Happy you thought of it first :)

Repo vs Controller: the user is something that's just for a specific request. the Repo lives based on what's configured in the IoC provider so if you cache it there you might end up with global state (TM). that leads to many tears for the developer as well as tears in the fabric of the space time continuum.